workspace/apps/pidp/src/app/features/access/pages/ms-teams/ms-teams.page.ts

```diff
@@ -76,14 +91,46 @@ export class MsTeamsPage
     this.currentPage++;
   }
 
-  private validateFirstPage(): boolean {
-    return this.checkValidity(
-      new FormArray([this.formState.clinicName, this.formState.clinicAddress])
-    );
+  public addClinicMember(): void {
+    const member = this.formState.buildClinicMemberForm();
+    this.formState.clinicMembers.push(member);
+  }
+
+  public removeClinicMember(index: number): void {
+    this.formState.clinicMembers.removeAt(index);
+  }
+
+  public getAgreementText(page: number): string {
+    return this.documentService.getMsTeamsAgreement(page);
+  }
+
+  public prepopulateForm(): void {
+    const partyId = this.partyService.partyId;
+
+    forkJoin({
+      personalInfo: this.personalInformationResource.get(partyId),
+      user: this.authorizedUserService.user$,
+    }).subscribe(({ personalInfo, user }) => {
+      this.clinicPrivacyOfficer.name = `${user.firstName} ${user.lastName}`;
+      this.clinicPrivacyOfficer.email = personalInfo?.email
+        ? personalInfo.email
+        : '';
+      this.clinicPrivacyOfficer.phone = personalInfo?.phone
+        ? personalInfo.phone
+        : '';
+      const privacyOfficer = {
+        name: this.clinicPrivacyOfficer.name,
```

Comment:
Since these are now in a combined `ForkJoin` I dont think you need the `clinicPrivacyOfficer ` local variable any more, you can just do the assignments here.